src/home/homework-luks.c

```diff
@@ -2110,6 +2110,21 @@ static int home_truncate(
         return !trunc; /* Return == 0 if we managed to truncate, > 0 if we managed to allocate */
 }
 
+static char **mkfs_options_for_fstype(const char *fstype) {
+        const char *e;
+        char *n;
+        char **r = NULL;
+
+        assert(fstype);
+
+        n = strjoina("SYSTEMD_HOME_MKFS_OPTIONS_", fstype);
+        e = getenv(ascii_strupper(n));
+        if (e)
+                r = strv_split(e, NULL);
+
+        return r;
```

Comment:
hmm, so this isn't pretty, because this means the function will return NULL in two distinct cases: in case of OOM, i.e. strv_split() can't allocate memory. And if the env var is not set.

These two cases should be handled differently by the caller though, one is a fatal thing (the oom), the other is totally fine.

Moreover, in our codebase `r` is the variable exclusively reserved for storing error codes in, i.e. it's always declared as `int r`. hence pleas rename this variable to something else.

maybe:

```c
static int mkfs_options_for_fstype(const char *fstype, char ***ret) {
        _cleanup_(strv_freep) char **l = NULL;
        const char *e;
        char *n;

        assert(fstype);

        n = strjoina("SYSTEMD_HOME_MKFS_OPTIONS_", fstype);
        e = getenv(ascii_strupper(n));
        if (e) {
                l = strv_split(e, NULL);
                if (!l)
                        return -ENOMEM;
        }

        *ret = TAKE_PTR(l);
        return 0;
}
```